src/p_mobj.c

```diff
@@ -532,6 +532,7 @@ static void P_ZMovement (mobj_t* mo)
 		// and utter appropriate sound.
 
 		mo->player->deltaviewheight = mo->momz>>3;
+		P_HitFloor (mo); // [FG] play sound when hitting animated floor
```

Comment:
What if we always play the splash sound like in Quake?
```diff
diff --git a/src/p_mobj.c b/src/p_mobj.c
index b88e931..d951bc6 100644
--- a/src/p_mobj.c
+++ b/src/p_mobj.c
@@ -523,8 +523,9 @@ floater:
 	    P_DamageMobj(mo, NULL, NULL, mo->health);
 	  else
 	    if (mo->player && // killough 5/12/98: exclude voodoo dolls
-		mo->player->mo == mo &&
-		mo->momz < -GRAVITY*8)
+		mo->player->mo == mo)
+	    {
+	      if (mo->momz < -GRAVITY*8)
 	      {
 		// Squat down.
 		// Decrease viewheight for a moment
@@ -532,9 +533,10 @@ floater:
 		// and utter appropriate sound.
 
 		mo->player->deltaviewheight = mo->momz>>3;
-		P_HitFloor (mo); // [FG] play sound when hitting animated floor
 		S_StartSound (mo, sfx_oof);
 	      }
+	      P_HitFloor (mo); // [FG] play sound when hitting animated floor
+	    }
 	  mo->momz = 0;
 	}
 
```
Not sure about this idea, Quake has swimming. In general it sounds a little strange without the sound of footsteps.

Reply:
In GZDoom, the splash sound is played instead of "oof", not in addition to it.

Reply:
> Not sure about this idea, Quake has swimming. In general it sounds a little strange without the sound of footsteps.

I think there is a separate set of sounds for the footsteps in the WAD that was provided in the thread. Will have to check.

> In GZDoom, the splash sound is played instead of "oof", not in addition to it.

Thanks for checking!

My general concern is multiplayer. If we replace the "oof" sound with the splash, then a player who hasn't the PWAD loaded won't hear anything if another player hits liquid ground. I think it should be an additional sound.

Reply:
> I doubt the water splash sound is a good idea.

Hm, too non-Vanilla? 🤔

Reply:
> Hm, too non-Vanilla? 🤔

Well, I guess it's ok as a secret feature. This seems wrong to me, because Doom doesn't have this type of sounds. Also, the player will make all these loud splashes, and the monsters still won't hear him 🤓

Reply:
Yes, it's a slittery slope. Doom didn't have brightmaps or the "secret revealed" message either, and yet I never play without them anymore. The question is where to stop adding things just because.

> Also, the player will make all these loud splashes, and the monsters still won't hear him

Well, if you can run a chainsaw behind their back without them noticing.. 😁

Reply:
Personally I like the sploosh noise in heretic when you fall into water. But this would feel out of place to me in doom.

Reply:
The sounds would only ever be heard if someone loads a PWAD which provides the corresponding lumps. And at that point I would say it happens in purpose and nobody should be surprised that it works. :wink: